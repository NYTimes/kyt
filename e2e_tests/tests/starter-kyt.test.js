const shell = require('shelljs');
const util = require('../fixtures/util');
const kill = require('../utils/psKill');
const path = require('path');
const fs = require('fs');

shell.config.silent = true;

describe('starter kyts', () => {
  describe('kyt-starter-universal', () => {
    beforeAll(() => {
      shell.cd(path.join(util.rootDir, 'packages/kyt-starter-universal'));
    });

    jasmine.DEFAULT_TIMEOUT_INTERVAL = 1000000; // eslint-disable-line no-undef

    it('should start a dev server on :3000', () => {
      let htmlOutput;
      const run = new Promise((resolve) => {
        const child = shell.exec('node_modules/kyt/cli/index.js dev', () => {
          resolve(htmlOutput);
        });
        child.stdout.on('data', (data) => {
          if (data.includes('✅  Development started')) {
            shell.exec('sleep 5');
            const output = shell.exec('curl -sb -o "" localhost:3000');
            htmlOutput = output.stdout;
            kill(child.pid);
          }
        });
      });
      return run.then(output => expect(output).toMatchSnapshot());
    });

    jasmine.DEFAULT_TIMEOUT_INTERVAL = 400000; // eslint-disable-line no-undef

    it('should build and run', () => {
      let htmlOutput;
      shell.exec('node_modules/kyt/cli/index.js build');
      const run = new Promise((resolve) => {
        const child = shell.exec('node build/server/main.js', () => {
          resolve(htmlOutput);
        });
        child.stdout.on('data', (data) => {
          if (data.includes('✅  server started on port: 3000')) {
            shell.exec('sleep 5');
            const output = shell.exec('curl -sb -o "" localhost:3000');
            htmlOutput = output.stdout;
            kill(child.pid);
          }
        });
      });
      return run.then(output => expect(output).toMatchSnapshot());
    });

    afterAll(() => {
      shell.rm('-rf', 'build');
      shell.cd(util.rootDir);
    });
  });

  describe('kyt-starter-static', () => {
    beforeAll(() => {
      shell.cd(path.join(util.rootDir, 'packages/kyt-starter-static'));
    });

    window.jasmine.DEFAULT_TIMEOUT_INTERVAL = 300000; // eslint-disable-line no-undef

    it('should start a server on :3001', () => {
      let htmlOutput;
      const run = new Promise((resolve) => {
        const child = shell.exec('node_modules/kyt/cli/index.js dev', () => {
          resolve(htmlOutput);
        });
        child.stdout.on('data', (data) => {
          if (data.includes('✅  Client started')) {
            shell.exec('sleep 5');
            const output = shell.exec('curl -sb -o "" localhost:3001');
            htmlOutput = output.stdout;
            kill(child.pid);
          }
        });
      });
      return run.then(output => expect(output).toMatchSnapshot());
    });

    it('should build', () => {
      const output = shell.exec('node_modules/kyt/cli/index.js build');
      expect(output.stdout.includes('✅  Done building')).toBe(true);
      const htmlOutput = fs.readFileSync('build/public/index.html', 'utf8');
      expect(htmlOutput).toMatchSnapshot();
    });

    afterAll(() => {
      shell.rm('-rf', 'build');
      shell.cd(util.rootDir);
    });
  });
});
