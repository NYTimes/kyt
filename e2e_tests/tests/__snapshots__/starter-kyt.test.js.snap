exports[`starter kyts kyt-starter-static should build 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta httpEquiv=\"X-UA-Compatible\" content=\"IE=edge\" />
    <meta charSet=\'utf-8\' />
    <meta httpEquiv=\"Content-Language\" content=\"en\" />
    <meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">
    <link id=\"favicon\" rel=\"shortcut icon\" href=\"/kyt-favicon.png\" sizes=\"16x16 32x32\" type=\"image/png\" />
    <link rel=\"stylesheet\" type=\"text/css\" href=\"https://cdnjs.cloudflare.com/ajax/libs/meyer-reset/2.0/reset.min.css\">
    <link rel=\"stylesheet\" type=\"text/css\" href=\"https://cdnjs.cloudflare.com/ajax/libs/normalize/4.2.0/normalize.min.css\">

    <title>Static kyt</title>
  <link href=\"/main-d85c35e4c326c7641389.css\" rel=\"stylesheet\"></head>

  <body>
    <div id=\"root\"></div>
  <script type=\"text/javascript\" src=\"/main-d85c35e4c326c7641389.js\"></script></body>
</html>
"
`;

exports[`starter kyts kyt-starter-static should start a server on :3001 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta httpEquiv=\"X-UA-Compatible\" content=\"IE=edge\" />
    <meta charSet=\'utf-8\' />
    <meta httpEquiv=\"Content-Language\" content=\"en\" />
    <meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">
    <link id=\"favicon\" rel=\"shortcut icon\" href=\"/kyt-favicon.png\" sizes=\"16x16 32x32\" type=\"image/png\" />
    <link rel=\"stylesheet\" type=\"text/css\" href=\"https://cdnjs.cloudflare.com/ajax/libs/meyer-reset/2.0/reset.min.css\">
    <link rel=\"stylesheet\" type=\"text/css\" href=\"https://cdnjs.cloudflare.com/ajax/libs/normalize/4.2.0/normalize.min.css\">

    <title>Static kyt</title>
  </head>

  <body>
    <div id=\"root\"></div>
  <script type=\"text/javascript\" src=\"http://localhost:3001/main.js\"></script></body>
</html>
"
`;

exports[`starter kyts kyt-starter-universal should build and run 1`] = `
"

<!DOCTYPE html>
<html lang=\"en\">

  <head>
    <meta httpEquiv=\"X-UA-Compatible\" content=\"IE=edge\" />
    <meta charSet=\'utf-8\' />
    <meta httpEquiv=\"Content-Language\" content=\"en\" />
    <meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">

    <link id=\"favicon\" rel=\"shortcut icon\" href=\"/kyt-favicon.png\" sizes=\"16x16 32x32\" type=\"image/png\" />
    <link rel=\"stylesheet\" type=\"text/css\" href=\"https://cdnjs.cloudflare.com/ajax/libs/meyer-reset/2.0/reset.min.css\">
    <link rel=\"stylesheet\" type=\"text/css\" href=\"https://cdnjs.cloudflare.com/ajax/libs/normalize/4.2.0/normalize.min.css\">
    <link rel=\"stylesheet\" type=\"text/css\" href=\"/main-874e1577c8b53897b468.css\">

    <title>Universal React Starter Kyt</title>
  </head>

  <body>
    <div id=\"root\"><div><div data-reactroot=\"\" data-reactid=\"1\" data-react-checksum=\"1154300769\"><i class=\"styles-logo--3UfME\" data-reactid=\"2\"></i><ul class=\"styles-nav--16eM0\" data-reactid=\"3\"><li class=\"styles-navItem--24uj4\" data-reactid=\"4\"><a class=\"styles-link--2TNqU\" href=\"/\" data-reactid=\"5\">Home</a></li><li class=\"styles-navItem--24uj4\" data-reactid=\"6\"><a class=\"styles-link--2TNqU\" href=\"/tools\" data-reactid=\"7\">Tools</a></li></ul><div class=\"styles-content--12E3U\" data-reactid=\"8\"><section data-reactid=\"9\"><p class=\"styles-paragraph--2F1Jd\" data-reactid=\"10\"><!-- react-text: 11 -->Welcome to the <!-- /react-text --><strong data-reactid=\"12\">Universal React Starter-kyt</strong><!-- react-text: 13 -->. This starter kyt should serve as the base for an advanced, server-rendered React app.<!-- /react-text --></p><p class=\"styles-paragraph--2F1Jd\" data-reactid=\"14\">Check out the Tools section for an outline of the libraries that are used in this Starter-kyt.</p></section></div></div></div></div>
    <script src=\"/main-874e1577c8b53897b468.js\"></script>
  </body>

</html>

"
`;

exports[`starter kyts kyt-starter-universal should start a dev server on :3000 1`] = `
"

<!DOCTYPE html>
<html lang=\"en\">

  <head>
    <meta httpEquiv=\"X-UA-Compatible\" content=\"IE=edge\" />
    <meta charSet=\'utf-8\' />
    <meta httpEquiv=\"Content-Language\" content=\"en\" />
    <meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">

    <link id=\"favicon\" rel=\"shortcut icon\" href=\"/kyt-favicon.png\" sizes=\"16x16 32x32\" type=\"image/png\" />
    <link rel=\"stylesheet\" type=\"text/css\" href=\"https://cdnjs.cloudflare.com/ajax/libs/meyer-reset/2.0/reset.min.css\">
    <link rel=\"stylesheet\" type=\"text/css\" href=\"https://cdnjs.cloudflare.com/ajax/libs/normalize/4.2.0/normalize.min.css\">
    

    <title>Universal React Starter Kyt</title>
  </head>

  <body>
    <div id=\"root\"><div><div data-reactroot=\"\" data-reactid=\"1\" data-react-checksum=\"1154300769\"><i class=\"styles-logo--3UfME\" data-reactid=\"2\"></i><ul class=\"styles-nav--16eM0\" data-reactid=\"3\"><li class=\"styles-navItem--24uj4\" data-reactid=\"4\"><a class=\"styles-link--2TNqU\" href=\"/\" data-reactid=\"5\">Home</a></li><li class=\"styles-navItem--24uj4\" data-reactid=\"6\"><a class=\"styles-link--2TNqU\" href=\"/tools\" data-reactid=\"7\">Tools</a></li></ul><div class=\"styles-content--12E3U\" data-reactid=\"8\"><section data-reactid=\"9\"><p class=\"styles-paragraph--2F1Jd\" data-reactid=\"10\"><!-- react-text: 11 -->Welcome to the <!-- /react-text --><strong data-reactid=\"12\">Universal React Starter-kyt</strong><!-- react-text: 13 -->. This starter kyt should serve as the base for an advanced, server-rendered React app.<!-- /react-text --></p><p class=\"styles-paragraph--2F1Jd\" data-reactid=\"14\">Check out the Tools section for an outline of the libraries that are used in this Starter-kyt.</p></section></div></div></div></div>
    <script src=\"http://localhost:3001/main.js\"></script>
  </body>

</html>

"
`;
